gcapi/gcapi.py

```diff
@@ -154,19 +158,15 @@ def detail(self, pk):
 class ModifiableMixin:
     _client = None  # type: Client
 
-    modify_json_schema = None  # type: jsonschema.Draft7Validator
-
     def __init__(self):
         pass
 
-    def _process_post_arguments(self, post_args):
-        if self.modify_json_schema is not None:
-            self.modify_json_schema.validate(post_args)
-
-    def _validate_data(self, data):
+    def _process_request_arguments(self, method, data):
         if data is None:
             data = {}
-        self._process_post_arguments(data)
+        schema = self.validation_schemas.get(method)
+        if schema:
+            schema.validate(data)
```

Comment:
Duplicate of `_verify_against_schema`?

Reply:
yes, but `_verify_against_schema` only deals with GET requests as inherited from the simpler base class. This function is different because it deals with all sorts of requests and is part of the `ModifiableMixin` interface. Since this is only a Mixin I cannot use `_verify_against_schema` directly because that function is not part of the class hierarchy and vice versa. Well, I could, but I think that is very messy at that point. Also, the `_process_request_arguments` function is overridden in some implementations to offer special functionality for POST/PATCH requests only. 

So this needs to remain separate somehow. Now that I refactored it is kind of obvious that both function are very similar and likely can be joined together somehow. I don't know. Make that call. Should I unify the two for just three lines of code? Probably is more code to implement that.

Reply:
Urgh, that Mixin stuff is a mess, instantiated incorrectly in all places (should be mixin first, then base class). Then you could use the base method. Is all of this covered by tests? None have been added in this PR. For some reason there is no report on codecov.

Reply:
The order did not really matter back in the days of python2. We still use python2 initialization so whatever. The entire Mixin stuff is used extensively  and covered by tests, yes. Actually we rely on it working, the integration tests are using it, for example. I did not change anything functionally when it comes to the feature set that is currently used just implemented if in a different way.  Example test:

https://github.com/DIAGNijmegen/rse-gcapi/blob/b00fb9f849fc8a39d6a90ff701addc7f2ea1f917/tests/integration_tests.py#L43-L44

---------------------

We do *not* test any PATCH requests however in our API as far as I can see. That require an awkward programmatical initialization of a reader-study, submitting some example answers and the PATCHing them. Quite messy to do as integration tests. A requests-mock based test might work however.But we do not have a requests-mock test in gcapi yet, so I would suggest splitting this out into another issue.

Reply:
I mean tests of the validation and what you have worked on in this PR is working, I cannot see any.

Reply:
That is pretty true. Like I said: Not PATCH request tests yet. I will see what I can do.